Sources/main.cpp

```diff
@@ -83,19 +84,35 @@ CreateSocket(std::string const &host, std::string const &port, bool reverse) {
 
 #if !defined(OS_WIN32)
 static int PlatformMain(std::string const &host, std::string const &port) {
-  std::unique_ptr<Socket> server = CreateSocket(host, port, false);
+  struct PlatformClient {
+    std::unique_ptr<Socket> socket;
+    PlatformSessionImpl impl;
+    Session session;
+
+    PlatformClient(std::unique_ptr<Socket> socket_)
+        : socket(std::move(socket_)),
+          session(ds2::GDBRemote::kCompatibilityModeLLDB) {
+      session.setDelegate(&impl);
+      session.create(socket.get());
+    }
+  };
 
-  PlatformSessionImpl impl;
+  std::unique_ptr<Socket> serverSocket = CreateSocket(host, port, false);
 
   do {
-    std::unique_ptr<Socket> client = server->accept();
-    // Platform mode implies that we are talking to an LLDB remote.
-    Session session(ds2::GDBRemote::kCompatibilityModeLLDB);
-    session.setDelegate(&impl);
-    session.create(client.get());
-
-    while (session.receive(/*cooked=*/false))
-      continue;
+    std::unique_ptr<Socket> clientSocket = serverSocket->accept();
+    auto platformClient =
+        ds2::make_unique<PlatformClient>(std::move(clientSocket));
+
+    std::thread thread(
+        [](std::unique_ptr<PlatformClient> client) {
+          while (client->session.receive(/*cooked=*/false)) {
+            continue;
+          }
+        },
+        std::move(platformClient));
+
+    thread.detach();
```

Comment:
Because it is easier to just create a new thread every time a client
connects. Doing otherwise would require polling and such.

On Monday, July 11, 2016, strager <notifications@github.com> wrote:

> In Sources/main.cpp
> <https://github.com/facebook/ds2/pull/403#discussion_r70359489>:
>
> > -
> > -    while (session.receive(/*cooked=*/false))
> > -      continue;
> > +    std::unique_ptr<Socket> clientSocket = serverSocket->accept();
> > +    auto platformClient =
> > +        ds2::make_unique<PlatformClient>(std::move(clientSocket));
> > +
> > +    std::thread thread(
> > +        [](std::unique_ptr<PlatformClient> client) {
> > +          while (client->session.receive(/*cooked=*/false)) {
> > +            continue;
> > +          }
> > +        },
> > +        std::move(platformClient));
> > +
> > +    thread.detach();
>
> Why are multiple threads preferred over handling multiple connections on a
> single thread?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/403/files/e6d0f35a1ebb85fe8c6e1d0e91b19a4d9aee6752..024d6577131834ae01bc46af0838d244b7588467#r70359489>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADHbJyI_Qr0G9mPuiDGpRKG2jV7phFEks5qUtwGgaJpZM4JIJxl>
> .
>


-- 
Stephane Sezer